### PR TITLE
Let FETools::get_fe_by_name return a std::unique_ptr

### PR DESCRIPTION
**Pull request by masterleinad**
Relates to #5880. We can omit some exception handling in `fe_tools.templates.h` otherwise it is pretty straightforward.

### CODE REVIEW

File: include/deal.II/fe/fe_tools.h [hunk 1]
```diff
@@ -1211,9 +1213,8 @@ namespace FETools
    * If no finite element can be reconstructed from this string, an exception
    * of type @p FETools::ExcInvalidFEName is thrown.
    *
-   * The function returns a pointer to a newly create finite element. It is in
-   * the caller's responsibility to destroy the object pointed to at an
-   * appropriate later time.
+   * The function returns a std::unique_ptr to a newly created finite element
+   * meaning the caller attains ownership over the returned object.
```
Comment (drwells) [inline include/deal.II/fe/fe_tools.h]: picky, but 'obtain' is more correct here.

Comment (masterleinad) [reply include/deal.II/fe/fe_tools.h]: Thanks!

**Comment by drwells** [comment]
Since this falls under the 'replace plain pointers with `std::unique_ptr`' umbrella I do not think that it should be at all controversial. I'll merge in a day or so unless someone objects.

**Pull request merged**